sycl/include/sycl/detail/util.hpp

```diff
@@ -67,6 +68,22 @@ struct CmpCStr {
 
 using SerializedObj = std::vector<unsigned char>;
 
+// We need special handling of std::string to handle ABI incompatibility
+// for get_info<>() when it returns std::string and vector<std::string>.
+// For this purpose, get_info_internal<>() is created to handle special
+// cases, and it is only called internally and not exposed to the user.
+// The following ReturnType structure is intended for general return type,
+// and special return types (std::string and vector of it).
+template <typename T> struct GetInfoReturnType { using type = T; };
```

Comment:
Non-blocking nitpick, but to be more accurate:
```suggestion
template <typename T> struct GetInfoImplReturnType { using type = T; };
```
Or I would actually prefer something more generalized, perhaps something like
```suggestion
template <typename T> struct ABINeutralT { using type = T; };
```